### PR TITLE
br: add error into retry list when backup

### PR DESCRIPTION
**Pull request by joccau**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/34865

Problem Summary:

### What is changed and how it works?
br need to retry backup region when br received the specific error from TiKV when TiKV write sst-file to storage.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by fengou1 — APPROVED** [review]

File: br/pkg/utils/retry.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ var retryableServerError = []string{
 	"error during dispatch",
 	"put object timeout",
 	"internalerror",
+	"not read from or written to within the timeout period",
```
Comment (3pointer) [inline br/pkg/utils/retry.go:27]: I think maybe we should add some retry even we meet the error not in the list.

Comment (joccau) [reply br/pkg/utils/retry.go:27]: For safety,  add this error into list currenty, I suggest.
But we add the logic(wait and retry the operation(backup and retry)) into master branch until the work is stable and safe.

**Review by 3pointer — APPROVED** [review]
LGTM

**Pull request merged**